### ISSUE TITLE
VST2: settings change when quickly doing a stop/play

### ISSUE THREAD

**Issue by pietro68** [audacity/audacity#4014]
**Steps to reproduce:**

- have some audio
- add DeeGain in the sidebar
- open its UI (the fader is at 0.0 dB)
- start playback by pressing Space
- move the fader down to -20.0 while playing the track
- hit the Space key twice to stop and play the track respectively

Actual result: the slider is moved back to 0.0 dB
Expected result: the slider remains at -20.0 dB

**Platforms**: Windows, MacOS

**Comment by pietro68** [comment]
I found that this issue is due to ::OnIdle not being reached before the rapid stop+play sequence, therefore not being able to send the accumulated slider changes to the framework. 

How I came to this conclusion: with the help of debug prints, I saw that the settings coming into VSTEffectInstance::ProcessInitialize were wrong (Gain=0.5) when the bug happened; I added a field "origin" to the settings, to see if the origin of the settings was different when the bug happened vs. when it did not happen.

The bug did not happen when the origin of the incoming settings was ::OnIdle, and it happened when the origin of the settings was ::MakeSettings. Excerpt of the log:

Start - Stop/Start good
### Gain=0.502 origin=1  (i.e. from MakeSettings)
### Gain=0.000 origin=5  (i.e. from OnIdle)

Start - Stop/Start BAD
### Gain=0.502 origin=1   (i.e. from MakeSettings)
### Gain=0.502 origin=1   (i.e. from MakeSettings)

Basically in the BAD case, the framework sends the initial settings it got (from MakeSettings) and never got an update of them by OnIdle.

The code to try this is found at this branch: https://github.com/pietro68/audacity/tree/debug-4014 just replace the windows print macro _RPT with whatever applies to MacOS.

**Comment by dozzzzer** [comment]
Verified on macOS Ventura & Windows 11
